### DISCUSSION TITLE
有点点小问题想问一下

### DISCUSSION THREAD

**Discussion by PxGame** [Molunerfinn/PicGo#702] [Q&A]
1、如何在代码中移除所有第三方图床
2、能否内置FTP插件的功能，现在使用FTP，还需要安装Nodejs，感觉很复杂

**Comment by Molunerfinn** [comment]
1. buildin的第三方图床无法移除。不太明白，如果已经用代码实现了，为何还需要移除？
2. 不再对内置图床做改动。就算你不用FTP，其实也需要node.js。另外Node.js安装只有两个步骤，1.官网下载安装包，2.无脑下一步。

**Reply by PxGame** [reply]
现在是在给项目组写wiki用，想缩减软件其他功能，减少软件配置环节。可以理解使用他的人得开箱即用，且不能使用别的图床

**Reply by Molunerfinn** [reply]
看你想直接改造PicGo还是使用更底层的PicGo-Core。

PicGo有个功能是显示想用的图床

![image](https://user-images.githubusercontent.com/12621342/126754323-bf0e1ed1-cb40-4a19-8460-b93ac0632899.png)
理论上配置了之后只能看见当前的图床：
![image](https://user-images.githubusercontent.com/12621342/126754345-4cce91cc-ba84-4414-8d6f-8028ccb199ce.png)
可以查看配置文件是如何变化的。

加个图床也不难，可以使用 `picgo.pluginLoader.registerPlugin` 实现

https://github.com/PicGo/PicGo-Core/blob/dev/src/lib/PluginLoader.ts#L65

![image](https://user-images.githubusercontent.com/12621342/126754609-3a29e2da-9dcf-47ca-8e45-b41ec03540ed.png)

**Reply by PxGame** [reply]
感谢大佬解答，大佬的工具挺好用，所以想直接修改。目前我把这些都移除了

![image](https://user-images.githubusercontent.com/7000730/126754925-871f080d-21cf-415e-bc03-65af983af2fb.png)
![image](https://user-images.githubusercontent.com/7000730/126754974-d02071d9-02a6-4efb-a728-c978b8c37acd.png)
![image](https://user-images.githubusercontent.com/7000730/126754995-fce93296-2bb1-4df1-ab12-859c63330d68.png)

不清楚，为啥界面上还有显示，且显示的字符，在代码中没有查找到。

![image](https://user-images.githubusercontent.com/7000730/126755161-089b5046-76d9-466a-a021-8a70c02d4a1a.png)

**Reply by Molunerfinn** [reply]
![image](https://user-images.githubusercontent.com/12621342/126755801-cb8960c9-1f25-40a5-9fd9-33e7add9d7c0.png)
剩下的代码需要你自己去读了，我没法帮你解释一遍

**Reply by PxGame** [reply]
好的，感谢！

**Reply by PxGame** [reply]
大佬，请问这个函数在哪实现的？
![image](https://user-images.githubusercontent.com/7000730/126762330-825d7378-d238-4eca-afc8-d197da6f90cd.png)

**Reply by Molunerfinn** [reply]
https://github.com/PicGo/PicGo-Core/blob/dev/src/lib/LifecyclePlugins.ts#L50

**Reply by PxGame** [reply]
谢谢！
